plugin/src/main/kotlin/com/ritense/plugin/autodeployment/PluginAutoDeploymentDto.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2015-2023 Ritense BV, the Netherlands.
+ *
+ *  Licensed under EUPL, Version 1.2 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" basis,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.ritense.plugin.autodeployment
+
+import com.fasterxml.jackson.databind.node.JsonNodeFactory
+import com.fasterxml.jackson.databind.node.ObjectNode
+import com.ritense.plugin.domain.PluginDefinition
+import java.util.UUID
+
+data class PluginAutoDeploymentDto(
+    val id: UUID?,
```

Comment:
Is there a use-case where the id is null? It only makes things difficult in the future if you want to add process-links when you left it empty.

Reply:
the only reason the id is there is to be able to use it somewhere else(i.e process links/other plugins) otherwise it would be autogenerated. autogeneration was the default so i left it nullable.